backend/app/Savor22b/Action/PlaceUserHouseAction.cs

```diff
@@ -39,103 +34,161 @@ public PlaceUserHouseAction(int villageID, int targetX, int targetY)
             [nameof(TargetY)] = TargetY.Serialize(),
         }.ToImmutableDictionary();
 
-    protected override void LoadPlainValueInternal(
-        IImmutableDictionary<string, IValue> plainValue)
-    {
-        VillageID = plainValue[nameof(VillageID)].ToInteger();
-        TargetX = plainValue[nameof(TargetX)].ToInteger();
-        TargetY = plainValue[nameof(TargetY)].ToInteger();
-    }
-
-    private bool isAbleToPlaceHouse(Village village, int targetX, int targetY)
+    private static Village GetVillage(int villageID)
     {
-        int halfWidth = (village.Width - 1) / 2;
-        int halfHeight = (village.Height - 1) / 2;
-
-        // halfWidth는 중심점을 기준으로 좌우 몇칸 설치가 가능한 지를 나타냄. 그러니 절댓값 targetX가 halfWidth보다 크면 설치 불가
-
-        if (Math.Abs(targetX) > halfWidth)
-        {
-            return false;
-        }
+        var village = CsvDataHelper.GetVillageByID(villageID);
 
-        if (Math.Abs(targetY) > halfHeight)
+        if (village == null)
         {
-            return false;
+            throw new InvalidVillageException("Invalid village ID");
         }
 
-        return true;
+        return village;
     }
 
-    public override IAccountStateDelta Execute(IActionContext ctx)
+    private static void ValidateReplaceUserHouse(
+        long currentBlock,
+        RelocationState? relocationState
```

Comment:
```suggestion
        RelocationState relocationState
```
nullable하지 않은 객체를 주는게 좋아보입니당

Reply:
요거는 nullable 한 것도 의도한거라 유지하도록 하겠습니다.

(오 프 라 인 으 로 얘 기 했 습 니 다)